heka/sandbox/filters/telemetry_decoder_view.lua

```diff
@@ -79,19 +80,17 @@ function process_message ()
                 return -1, "Decoder is missing its name"
             end
 
-            local id = string.match(v.Name, "^TelemetryKafkaInput(%d+)")
-            if not id then
-                return -1, "Telemetry decoder is missing its identifier"
-            else
+            local id = string.match(v.Name, decoder_match)
+            if id then
                 id = tonumber(id)
-            end
 
-            if type(v["ProcessMessageCount-TelemetryDecoder"]) == "table" then
-                update_delta(ts, TOTAL, id, id_count, v["ProcessMessageCount-TelemetryDecoder"].value)
-            end
+                if type(v["ProcessMessageCount-TelemetryDecoder"]) == "table" then
+                    update_delta(ts, TOTAL, id, id_count, v["ProcessMessageCount-TelemetryDecoder"].value)
+                end
 
-            if type(v["ProcessMessageFailures-TelemetryDecoder"]) == "table" then
-                update_delta(ts, FAILURES, id, id_failures, v["ProcessMessageFailures-TelemetryDecoder"].value)
+                if type(v["ProcessMessageFailures-TelemetryDecoder"]) == "table" then
+                    update_delta(ts, FAILURES, id, id_failures, v["ProcessMessageFailures-TelemetryDecoder"].value)
+                end
             end
```

Comment:
Do you not care about the case where "id" is missing now? Or should it still have an "else return -1, ..." bit here?